mixing/mixpool/mixpool.go

```diff
@@ -1394,6 +1395,13 @@ func (p *Pool) checkAcceptKE(ke *wire.MsgMixKeyExchange) error {
 		return ruleError(ErrPeerPositionOutOfBounds)
 	}
 
+	now := time.Now()
+	keEpoch := time.Unix(int64(ke.Epoch), 0)
+	if now.Before(keEpoch) && keEpoch.Sub(now) > earlyKEDuration {
```

Comment:
Since the epoch is already in unix time, I'd suggest just doing the calculation in unix time.  It's more efficient and you don't have to worry about potential future precision issues from downcasting a `uint64` to an `int64` to boot.

```Go
const earlyKESecs = 5
nowUnix := time.Now().Unix()
if ke.Epoch + earlyKESecs <= uint64(nowUnix) {
	// error
}
```

Reply:
i lowkey hate this but if we are going to eventually move to network time anyways we are stuck with at least a second of precision loss anyways.

Reply:
this suggested change is incorrect and i've reverted it.

we can send KEs over the entire duration until the next epoch occurs, as sessions are recreated (for various reasons).  This check would reject those later KE messages that are received 5s after the current epoch.

Reply:
Already approved, but for future reference when network time is introduced, since that is the desired behavior, I'm pretty sure the check would be the following instead:

```Go
uint64(nowUnix) + earlyKESecs < ke.Epoch
```

To check that, we can use symbolic manipulation on the inequalities:

Set `a = now`, `b = epoch`, `c = threshold (earlyKESecs)`
Then, the code is `a < b && b-a > c`
```
=> a < b && b > c+a
=> a < b && a+c < b
```
Since `c>0` (aka is positive) if `a+c < b`, then `a`, which is smaller than `a+c` is definitely also smaller than `b`, thus the first check is redundant.
```
=> a+c < b
=> now+earlyKESecs < epoch
```